### PR TITLE
Feature/get vendor list

### PR DESCRIPTION
**Pull request by alextremp**
## Description

This PR includes the [getVendorList](https://github.com/InteractiveAdvertisingBureau/GDPR-Transparency-and-Consent-Framework/blob/master/TCFv2/IAB%20Tech%20Lab%20-%20CMP%20API%20v2.md#getvendorlist) implementation, for both the __tcfapi command consumer and the BorosTcf facade

image

## Solves ticket/s

* https://jira.scmspain.com/browse/PSP-3173

## Expected behavior

**window.__tcfapi('getVendorList', 2, callback, vendorListVersion)**
- will request the untranslated LATEST vendor list if no version is specified
- will request the untranslated specific vendorListVersion if version is specified
- will only return success=true with a valid response in the callback if
  - the vendorListVersion is not specified or it is LATEST or a valid number
  - the vendor list can be obtained 
- otherwise it will return success=false with a null response

**BorosTcf#getVendorList({version, language})**
- will request the untranslated LATEST vendor list if no version and no language are specified
- will request the untranslated specific version if version is specified but language is not
- will request the translated LATEST version if language is specified but version is not
- will request the translated specific version if both version and language are specified
- will only resolve a valid response if inputs are not defined or version is LATEST or a valid number and if language is a valid 2 chars lowercase language code
- otherwise it will reject 

Both calls will only perform a Http request if the requested vendor list was not requested before, according to the requested version and language specifications.

## Review steps

* check the GetVendorListCommandTest and BorosTcfTest specifications
* can be started in local to check the behavior in a browser's console

## Further considerations

just for a mention: 
- this PR includes the usage of the 'adapter' feature of the brusc IOC to instantiate a "cached vendor list repository" over the HttpVendorListRepository
- this PR includes the 'chain' feature of the brusc IOC to ease the test developments manipulating the HttpClient instance to allow us to just mock it and let all the code behave the same as it was a real execution, so no need to do granular testing, just check that the code is covered 

this PR as a little bigger than it could be because of some "testable" infrastructure implementation to ease future development

## Memetized description

mandatory

### CODE REVIEW

File: src/main/domain/vendorlist/Language.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      throw new InvalidLanguageValueError(languageValue)
+    }
+    if (languageValue.length !== 2) {
+      throw new InvalidLanguageValueError(languageValue)
+    }
+    return languageValue
+  }
+}
+
+const DEFAULT_LANGUAGE_VALUE = 'en'
```
Comment (maryammdot) [inline src/main/domain/vendorlist/Language.js:26]: maybe we should use as default spanish? Or is it a specification of IAB that the default is english?

File: src/main/infrastructure/repository/HttpVendorListRepository.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+            version: value.vendorListVersion,
+            language: language || DEFAULT_VENDOR_LIST_VALUE_LANGUAGE,
+            value
+          })
+      )
+  }
+}
+
+const VENDOR_LIST_ENDPOINT = 'https://a.dcdn.es/borostcf/v2/vendorlist'
+const DEFAULT_VENDOR_LIST_VALUE_LANGUAGE = 'en'
```
Comment (maryammdot) [inline src/main/infrastructure/repository/HttpVendorListRepository.js:34]: wouldn't be better to have this as a domain constant, now is repeated in two places, in the Language class and here.

File: src/main/infrastructure/repository/http/HttpClient.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+/**
+ * @interface
+ */
+class HttpClient {
+  post({url, data, options}) {}
```
Comment (maryammdot) [inline src/main/infrastructure/repository/http/HttpClient.js:5]: Why do we need in the interface the post? If the AxiosHttpClient doesn't implement it?

File: src/main/domain/vendorlist/Language.js (hunk 1 above)
Comment (alextremp) [reply src/main/domain/vendorlist/Language.js:26]: In the API side, the default is English and this is prepared in our backend to not call for any purpose-xx translation (in fact, the purposes-en.json does not exist, so language=en just returns the original vendorlist.json)

File: src/main/infrastructure/repository/HttpVendorListRepository.js (hunk 2 above)
Comment (alextremp) [reply src/main/infrastructure/repository/HttpVendorListRepository.js:34]: well seen, will refactor 👍

File: src/main/infrastructure/repository/http/HttpClient.js (hunk 3 above)
Comment (alextremp) [reply src/main/infrastructure/repository/http/HttpClient.js:5]: well seen, that's a copy-paste bug xD will change

**Comment by alextremp** [comment]
updated @maryammdot

File: src/main/application/BorosTcf.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import {inject} from '../core/ioc/ioc'
 
-import {GetVendorListUseCase} from './services/vendor_list/GetVendorListUseCase'
+import {GetVendorListUseCase} from './services/vendorlist/GetVendorListUseCase'
 import {GetConsentStatusUseCase} from './services/vendor_consent/GetConsentStatusUseCase'
```
Comment (oscarraigscmspain) [inline src/main/application/BorosTcf.js:4]: In BorosCmp the services were called 
vendor_list and vendor_consent. 
https://github.com/scm-spain/boros-CMP/tree/master/src/cmp/application/services.

Comment (alextremp) [reply src/main/application/BorosTcf.js:4]: I see (didn't look at the old repo at all), but applying here what we do at other projects, can be discussed :)

**Pull request merged**